### ISSUE TITLE
Feature Request: ForwardAuth Auth Plugin

### ISSUE THREAD

**Issue by dstar4138** [splitbrain/dokuwiki#3402]
TL;DR: Wrote a simple working implementation here, for the impatient:
https://gist.github.com/dstar4138/754c384d086d0b07ba3b072acc8f3957

**Feature Request:**

Support using ForwardAuth SSO headers provided by a reverse-proxy middleware. An example implementation of this is Authelia + Traefik [1]. This helps with providing stronger authentication primitives instead of having to hand-roll them within DokuWiki. The functionality inspects HTTP Headers like `Remote-User` and uses them as the user-data.

**Risks**

This isn't a perfect solution; and in production it may be advantageous to just use the LDAP Auth already provided. Additionally, Authelia is improving quickly and may eventually support OAuth2 [2], which would be safer, but obviates this implementation.

**Notes**

I don't really care about attribution; I'm not a PHP programmer. So use it how you wish. Regrettably, I don't have the cycles in the day to maintain this contribution if accepted. I just wanted this feature so I wrote a quick mock-up that functions enough for my homelab. I'm adding all the keywords I used when searching here so others may be able to find it. Hope it helps.

Thank you so much for your work and contributions to DokuWiki, it is excellent!

---

[1] - https://www.authelia.com/docs/deployment/supported-proxies/
[2] - https://github.com/authelia/authelia/issues/189

**Comment by phy25** [comment]
This looks neat but I am not sure if this should be in core; you are always welcome to host plugins yourself: https://www.dokuwiki.org/devel:plugins
